astropy/config/configuration.py

```diff
@@ -570,6 +577,68 @@ def get_config(packageormod=None, reload=False, rootname=None):
         return cobj
 
 
+def generate_config(pkgname='astropy', filename=None):
+    """Generates a configuration file, from the list of `ConfigItem`
+    objects for each subpackage.
+
+    .. versionadded:: 4.1
+
+    Parameters
+    ----------
+    packageormod : str or None
+        The package for which to retrieve the configuration object.
+    filename : str or file object or None
+        If None, the default configuration path is taken from `get_config`.
+
+    """
+    package = importlib.import_module(pkgname)
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore', AstropyDeprecationWarning)
+        for mod in pkgutil.walk_packages(path=package.__path__,
+                                         prefix=package.__name__ + '.'):
+
+            if mod.module_finder.path.endswith('tests'):
+                # Skip test modules
+                continue
+            if mod.name.split('.')[-1].startswith('_'):
+                # Skip private modules
+                continue
+
+            with contextlib.suppress(ImportError):
+                importlib.import_module(mod.name)
+
+    wrapper = TextWrapper(initial_indent="## ", subsequent_indent='## ',
+                          width=78)
+
+    if filename is None:
+        filename = get_config(package).filename
+
+    with contextlib.ExitStack() as stack:
+        if isinstance(filename, (str, pathlib.Path)):
+            fp = stack.enter_context(open(filename, 'w'))
+        else:
+            # assume it's a file object, or io.StringIO
+            fp = filename
+
+        # Parse the subclasses, ordered by their module name
+        subclasses = ConfigNamespace.__subclasses__()
+        for conf in sorted(subclasses, key=lambda x: x.__module__):
+            print_module = True
+            for item in conf.__dict__.values():
+                if not isinstance(item, ConfigItem):
+                    continue
+                if print_module:
+                    # If this is the first item of the module, we print the
+                    # module name, but not if this is the root package...
+                    if item.module != pkgname:
+                        modname = item.module.replace(f'{pkgname}.', '')
+                        fp.write(f"[{modname}]\n\n")
```

Comment:
Is it safer to use `os.linesep` instead of `"\n"`?

Reply:
`open` takes care of this by default:
> * On input, if newline is None, universal newlines mode is
  enabled. Lines in the input can end in '\n', '\r', or '\r\n', and
  these are translated into '\n' before being returned to the
  caller. [...]
> * On output, if newline is None, any '\n' characters written are
>  translated to the system default line separator, os.linesep.

Reply:
Ah, that's so good to know!